controllers/controllers/workloads/integration/cforg_controller_integration_test.go

```diff
@@ -0,0 +1,137 @@
+			Expect(k8sClient.Create(testCtx, &hncv1alpha2.HierarchyConfiguration{
+				ObjectMeta: metav1.ObjectMeta{Namespace: orgGUID, Name: "hierarchy"},
+			})).To(Succeed())
+		})
+
+		JustBeforeEach(func() {
+			Expect(k8sClient.Create(testCtx, &cfOrg)).To(Succeed())
+		})
+
+		It("creates a subnamespace anchor in the root namespace", func() {
+			var anchor hncv1alpha2.SubnamespaceAnchor
+			Eventually(func() error {
+				return k8sClient.Get(testCtx, types.NamespacedName{Namespace: rootNamespace.Name, Name: orgGUID}, &anchor)
+			}).Should(Succeed())
+
+			Expect(anchor.Labels).To(HaveKeyWithValue(workloads.OrgNameLabel, orgName))
+		})
+
+		It("makes the CFOrg the owner of its subnamespace anchor", func() {
+			var anchor hncv1alpha2.SubnamespaceAnchor
+			Eventually(func() error {
+				return k8sClient.Get(testCtx, types.NamespacedName{Namespace: rootNamespace.Name, Name: orgGUID}, &anchor)
+			}).Should(Succeed())
+
+			Expect(anchor.ObjectMeta.OwnerReferences).To(ConsistOf([]metav1.OwnerReference{
+				{
+					APIVersion: "workloads.cloudfoundry.org/v1alpha1",
+					Kind:       "CFOrg",
+					Name:       cfOrg.Name,
+					UID:        cfOrg.GetUID(),
+				},
+			}))
+		})
+
+		When("the subnamespace anchor status changes to ok", func() {
+			JustBeforeEach(func() {
+				var subspaceAnchor hncv1alpha2.SubnamespaceAnchor
+				Eventually(func() error {
+					return k8sClient.Get(testCtx, types.NamespacedName{Namespace: rootNamespace.Name, Name: orgGUID}, &subspaceAnchor)
+				}).Should(Succeed())
+				subspaceAnchor.Status.State = hncv1alpha2.Ok
+				Expect(k8sClient.Update(testCtx, &subspaceAnchor)).To(Succeed())
+			})
+
+			It("sets the AllowCascadingDeletion property on the HNC hierarchyconfiguration in the namespace", func() {
+				Eventually(func() (bool, error) {
+					var hc hncv1alpha2.HierarchyConfiguration
+					err := k8sClient.Get(testCtx, types.NamespacedName{Namespace: orgGUID, Name: "hierarchy"}, &hc)
+					return hc.Spec.AllowCascadingDeletion, err
+				}).Should(BeTrue())
+			})
+
+			It("sets the CFOrg 'Ready' condition to 'True'", func() {
+				Eventually(func() (bool, error) {
+					var latestOrg workloadsv1alpha1.CFOrg
+					err := k8sClient.Get(testCtx, types.NamespacedName{Namespace: rootNamespace.Name, Name: orgGUID}, &latestOrg)
+					return meta.IsStatusConditionTrue(latestOrg.Status.Conditions, "Ready"), err
+				}, 5*time.Second).Should(BeTrue())
+			})
+
+			It("sets the CFOrg status GUID", func() {
+				Eventually(func() (string, error) {
+					var latestOrg workloadsv1alpha1.CFOrg
+					err := k8sClient.Get(testCtx, types.NamespacedName{Namespace: rootNamespace.Name, Name: orgGUID}, &latestOrg)
+					return latestOrg.Status.GUID, err
+				}, 5*time.Second).Should(Equal(cfOrg.ObjectMeta.Name))
+			})
+		})
+
+		When("the subnamespace anchor never becomes ready", func() {
+			It("leaves the CFOrg 'Ready' condition as 'Unknown'", func() {
+				Eventually(func() (bool, error) {
```

Comment:
The function in `Eventually` and `Consistently` is the same, therefore could be extracted. Up to you really.